tools/table_compute/scripts/safety.py

```diff
@@ -0,0 +1,343 @@
+        'log', 'log10', 'log1p', 'log2', 'modf', 'nan',
+        'pi', 'pow', 'radians', 'remainder', 'sin', 'sinh',
+        'sqrt', 'tan', 'tanh', 'tau', 'trunc',
+        # Pandas DataFrame
+        'abs', 'add', 'agg', 'aggregate', 'align', 'all', 'any', 'append',
+        'apply', 'applymap', 'as_matrix', 'asfreq', 'at', 'axes', 'bool',
+        'boxplot', 'clip', 'clip_lower', 'clip_upper', 'columns', 'combine',
+        'compound', 'corr', 'count', 'cov', 'cummax', 'cummin', 'cumprod',
+        'cumsum', 'div', 'divide', 'dot', 'drop', 'drop_duplicates',
+        'droplevel', 'dropna', 'duplicated', 'empty', 'eq', 'equals',
+        'expanding', 'ffill', 'fillna', 'filter', 'first', 'first_valid_index',
+        'floordiv', 'ge', 'groupby', 'gt', 'head', 'hist', 'iloc', 'index',
+        'insert', 'interpolate', 'isin', 'isna', 'isnull', 'items', 'iteritems',
+        'iterrows', 'itertuples', 'ix', 'join', 'keys', 'kurt', 'kurtosis',
+        'last', 'last_valid_index', 'le', 'loc', 'lookup', 'lt', 'mad',
+        'mask', 'max', 'mean', 'median', 'melt', 'merge', 'min', 'mod',
+        'mode', 'mul', 'multiply', 'ndim', 'ne', 'nlargest', 'notna', 'notnull',
+        'nsmallest', 'nunique', 'pct_change', 'pivot', 'pivot_table', 'plot',
+        'pop', 'pow', 'prod', 'product', 'quantile', 'radd', 'rank', 'rdiv',
+        'replace', 'resample', 'rfloordiv', 'rmod', 'rmul', 'rolling', 'round',
+        'rpow', 'rsub', 'rtruediv', 'sample', 'select',
+        'sem', 'shape', 'shift', 'size', 'skew', 'slice_shift',
+        'squeeze', 'stack', 'std', 'sub', 'subtract', 'sum', 'swapaxes',
+        'swaplevel', 'tail', 'transform', 'transpose', 'truediv', 'truncate',
+        'tshift', 'style', 'take', 'unstack', 'var', 'where',
+        # Pandas
+        'DataFrame', 'array', 'concat', 'cut', 'date_range', 'datetime',
+        'factorize', 'interval_range', 'isna', 'isnull', 'melt', 'merge',
+        'notna', 'notnull', 'np', 'period_range', 'pivot', 'pivot_table',
+        'plotting', 'unique', 'value_counts', 'wide_to_long'
+    )
+
+    def __init__(self, obj_name, custom_string, strict_functiondefs):
+        if isinstance(obj_name, list):
+            self.name = obj_name[0]
+            self.allowed_names = obj_name + list(self.__allowed)
+        else:
+            self.name = obj_name
+            self.allowed_names = list(self.__allowed)
+
+        self.expr = custom_string
+        self.parseXML(strict_functiondefs)
+        self.__assertSafe()
+
+    def parseXML(self, xml_file):
+        root = ET.parse(xml_file).getroot()
+        options = [option.attrib["value"]
+                   for macro in root for option in macro
+                   if macro.tag == "macro" and option.tag == "option"]
+
+        self.allowed_names = self.allowed_names + options
+
+    def generateMultiLine(self):
+        """
+        Generates a multi-line function to be evaluated outside the class
+        """
+        cust_fun = "def fun(%s):%s" % (
```

Comment:
```suggestion
        cust_fun = "def fun(%s):\n%s" % (
```

not sure, but isn't your version failing to introduce the first line break?